File: modules/core/misc/java/test/SizeTest.java
package org.opencv.test.core;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import org.junit.Test;
import org.opencv.core.Point;
import org.opencv.core.Size;
import org.opencv.test.OpenCVTestCase;

public class SizeTest extends OpenCVTestCase {

    Size dstSize;
    Size sz1;
    Size sz2;

    @Override
    public void setUp() throws Exception {
        super.setUp();

        sz1 = new Size(10.0, 10.0);
        sz2 = new Size(-1, -1);
        dstSize = null;
    }

    @Test
    public void testArea() {
        double area = sz1.area();
        assertEquals(100.0, area, 0);
    }

    @Test
    public void testClone() {
        dstSize = sz1.clone();
        assertEquals(sz1, dstSize);
    }

    @Test
    public void testEqualsObject() {
        assertFalse(sz1.equals(sz2));

        sz2 = sz1.clone();
        assertTrue(sz1.equals(sz2));
    }

    @Test
    public void testHashCode() {
        assertEquals(sz1.hashCode(), sz1.hashCode());
    }

    @Test
    public void testSet() {
        double[] vals1 = {};
        sz2.set(vals1);
        assertEquals(0., sz2.width, 0);
        assertEquals(0., sz2.height, 0);

        double[] vals2 = { 9, 12 };
        sz1.set(vals2);
        assertEquals(9., sz1.width, 0);
        assertEquals(12., sz1.height, 0);
    }

    @Test
    public void testSize() {
        dstSize = new Size();

        assertNotNull(dstSize);
        assertEquals(0., dstSize.width, 0);
        assertEquals(0., dstSize.height, 0);
    }

    @Test
    public void testSizeDoubleArray() {
        double[] vals = { 10, 20 };
        sz2 = new Size(vals);

        assertEquals(10., sz2.width, 0);
        assertEquals(20., sz2.height, 0);
    }

    @Test
    public void testSizeDoubleDouble() {
        assertNotNull(sz1);

        assertEquals(10.0, sz1.width, 0);
        assertEquals(10.0, sz1.height, 0);
    }

    @Test
    public void testSizePoint() {
        Point p = new Point(2, 4);
        sz1 = new Size(p);

        assertNotNull(sz1);
        assertEquals(2.0, sz1.width, 0);
        assertEquals(4.0, sz1.height, 0);
    }

    @Test
    public void testToString() {
        String actual = sz1.toString();
        String expected = "10x10";
        assertEquals(expected, actual);
    }

}

File: modules/core/misc/objc/test/SizeTest.swift
//
//  SizeTest.swift
//
//  Created by Giles Payne on 2020/01/31.
//

import XCTest
import OpenCV

class SizeTest: OpenCVTestCase {

    let sz1 = Size2d(width: 10.0, height: 10.0)
    let sz2 = Size2d(width: -1, height: -1)

    func testArea() {
        let area = sz1.area()
        XCTAssertEqual(100.0, area);
    }

    func testClone() {
        let dstSize = sz1.clone()
        XCTAssertEqual(sz1, dstSize)
    }

    func testEqualsObject() {
        XCTAssertFalse(sz1 == sz2);

        let sz2 = sz1.clone();
        XCTAssertTrue(sz1 == sz2);
    }

    func testHashCode() {
        XCTAssertEqual(sz1.hash(), sz1.hash());
    }

    func testSet() {
        let vals1:[Double] = []
        sz2.set(vals: vals1 as [NSNumber])
        XCTAssertEqual(0, sz2.width);
        XCTAssertEqual(0, sz2.height);

        let vals2:[Double] = [9, 12]
        sz1.set(vals: vals2 as [NSNumber]);
        XCTAssertEqual(9, sz1.width);
        XCTAssertEqual(12, sz1.height);
    }

    func testSize() {
        let dstSize = Size2d()

        XCTAssertNotNil(dstSize)
        XCTAssertEqual(0, dstSize.width)
        XCTAssertEqual(0, dstSize.height)
    }

    func testSizeDoubleArray() {
        let vals:[Double] = [10, 20]
        let sz2 = Size2d(vals: vals as [NSNumber])

        XCTAssertEqual(10, sz2.width)
        XCTAssertEqual(20, sz2.height)
    }

    func testSizeDoubleDouble() {
        XCTAssertNotNil(sz1)

        XCTAssertEqual(10.0, sz1.width)
        XCTAssertEqual(10.0, sz1.height)
    }

    func testSizePoint() {
        let p = Point2d(x: 2, y: 4)
        let sz1 = Size2d(point: p)

        XCTAssertNotNil(sz1)
        XCTAssertEqual(2.0, sz1.width)
        XCTAssertEqual(4.0, sz1.height)
    }

    func testToString() {
        let actual = "\(sz1)"
        let expected = "Size2d {10.000000,10.000000}"
        XCTAssertEqual(expected, actual);
    }

}
